client/data/agency-dashboard/use-sync-jetpack-connection-status.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { useContext, useEffect, useMemo } from 'react';
+import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
+import wpcom from 'calypso/lib/wp';
+
+const useSyncJetpackConnectionStatus = (
+	isPartnerOAuthTokenLoaded: boolean,
+	isConnectionHealthy: boolean,
+	siteId: number
+) => {
+	const queryClient = useQueryClient();
+	const { search, currentPage, filter, sort } = useContext( SitesOverviewContext );
+
+	const dashboardSitesQueryKey = useMemo(
```

Comment:
Also, what do we really earn by memoizing it?

Reply:
It's usable in the dependency array of the `useEffect` call later in the hook.